### PR TITLE
refactor(ecs): remove duplication with CloudMap and ECS NameSpace

### PR DESCRIPTION
**Pull request by iamhopaul123**
----
Remove duplication/confusion with CloudMap and ECS namespaceType by substituting the ECS namespaceType with CloudMap namespaceType.

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [N/A] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by iamhopaul123** [comment]
@pkandasamy91 - please review

File: packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ export = {
     // WHEN
     cluster.addDefaultCloudMapNamespace({
       name: "foo.com",
-      type: ecs.NamespaceType.PUBLIC_DNS
+      type: cloudmap.NamespaceType.DNS_PRIVATE
```
Comment (pkandasamy91) [inline packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts]: Shouldn't this be DNS_PUBLIC? Does this unit test actually pass? If so we might want to add a test to verify this.

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export class Cluster extends Resource implements ICluster {
       throw new Error("Can only add default namespace once.");
     }
 
-    const namespaceType = options.type === undefined || options.type === NamespaceType.PRIVATE_DNS
+    const namespaceType = options.type === undefined
```
Comment (pkandasamy91) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts]: If options.type is PRIVATE_DNS, then you'll actually be setting the namespaceType to DNS_PUBLIC even though it should be private. You can either leave it as is or change it to something like:

const namespaceType = options.type !== undefined 
? options.type
: cloudmap.NamespaceType.DNS_PRIVATE

File: packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts (hunk 1 above)
Comment (iamhopaul123) [reply packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts]: Yes, it passes the unit test.

Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts]: Then can you also have this check fail if the namespace type isn't what's passed in?

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 2 above)
Comment (iamhopaul123) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts]: Yes, I agree with you. Nice catch!

**Pull request merged**